src/Servers/Kestrel/perf/Microbenchmarks/Http2/Http2ConnectionBenchmarkBase.cs

```diff
@@ -56,6 +53,16 @@ public virtual void GlobalSetup()
         _httpRequestHeaders[HeaderNames.Scheme] = new StringValues("http");
         _httpRequestHeaders[HeaderNames.Authority] = new StringValues("localhost:80");
 
+        if (NumCookies > 0)
+        {
+            var cookies = new StringValues("0=1");
+            for (var index = 1; index < NumCookies; index++)
+            {
+                cookies = cookies.Append($"{index}={index + 1}").ToArray();
```

Comment:
Found it:
StringValues.Concat
https://github.com/dotnet/runtime/blob/e8a18e2cd5d03f7b1801595d3effcc52082ffb64/src/libraries/Microsoft.Extensions.Primitives/src/StringValues.cs#L476
```suggestion
                cookies = cookies.Concat($"{index}={index + 1}");
```

Reply:
`Concat` for `StringValues` takes 2 arguments (https://source.dot.net/#Microsoft.Extensions.Primitives/StringValues.cs,449). It ends up quite ugly with either of the 3 overrides. So, I reverted to the `.ToArray()` that Stephen suggested

Reply:
I like @Tratcher's solution better once updated to correctly call it as a static method. Neither is very efficient though.

```suggestion
                cookies = StringValues.Concat(cookies, $"{index}={index + 1}");
```

The most efficient thing would be to pre-allocate the array.

```csharp
var cookies = new string?[NumCookies];
for (var index = 1; index < NumCookies; index++)
{
    cookies[index] = $"{index}={index + 1}";
}
_httpRequestHeaders[HeaderNames.Cookie] = cookies;
```

Reply:
Ok sounds good. If we are worrying about performance here though, should we also remove the unnecessary plus operation, or is that overkill?

```C#
var cookies = new string[NumCookies];
for (var index = 1; index < NumCookies;)
{
    cookies[index] = $"{index}={++index}";
}
_httpRequestHeaders[HeaderNames.Cookie] = cookies;
```

Reply:
I'm not worried about performance in the setup part of the benchmark, but reallocating an array on each iteration of a loop is a little bit beyond the pale under any circumstances in my opinion. We have standards! 😄  Especially in a case like this where the more efficient code is also the more concise easy-to-read version.

Even if this was something that got called in production for every single request like `MergeCookies()` does, I would not contort the for-loop just to save a single add operation per iteration. Those are practically free. At least not without strong evidence that it made a measurable performance impact relative to everything else that happens in order to process a request.

If you really wanted to optimize this, you could use the same string for every cookie value. Kestrel doesn't do any validation. But like I said, I'm not worried about performance in the setup part of the benchmark.

Reply:
Ok makes sense. Thanks for explaining.